### PR TITLE
Fail codegen "validation" if it caused a change

### PR DESCRIPTION
**Pull request by mkolesnik**
If the repo is dirty after running the validation, fail as it indicates
that the API changed and needs to ge updated.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -16,6 +16,8 @@ jobs:
         uses: actions/checkout@master
       - name: Run validation
         run: make codegen
+      - name: Make sure there's no pending changes
+        run: output=$(git status --porcelain) && [ -z "$output" ]
```
Comment (skitt) [inline .github/workflows/linting.yml:20]:
```suggestion
        run: git diff --exit-code
```

This will show the differences in the logs too.

**Review by skitt — CHANGES_REQUESTED** [review]

File: .github/workflows/linting.yml (hunk 1 above)
Comment (mkolesnik) [reply .github/workflows/linting.yml:20]: Problem is that `git diff` doesnt count untracked files, so if such appear we'll miss them, while `git status` does

File: pkg/apis/submariner.io/v1/types.go [hunk 2]
```diff
@@ -46,6 +46,7 @@ type EndpointSpec struct {
 	ClusterID     string            `json:"cluster_id"`
 	CableName     string            `json:"cable_name"`
 	Hostname      string            `json:"hostname"`
+	Hostname2     string            `json:"hostname2"`
```
Comment (dfarrell07) [inline pkg/apis/submariner.io/v1/types.go:49]: I think these type changes shouldn't be a part of this commit?

**Review by dfarrell07 — CHANGES_REQUESTED** [review]

File: .github/workflows/linting.yml (hunk 1 above)
Comment (skitt) [reply .github/workflows/linting.yml:20]: Good point, so

```suggestion
        run: git add -A && git diff --staged --exit-code
```

File: pkg/apis/submariner.io/v1/types.go (hunk 2 above)
Comment (skitt) [reply pkg/apis/submariner.io/v1/types.go:49]: I imagine the point here was to prove that the test works ;-). (But yes, they’ll need to be removed before the merge.)

**Comment by dfarrell07** [comment]
Closes: https://github.com/submariner-io/submariner/issues/809

**Comment by dfarrell07** [comment]
Maybe we can finish this one off @mkolesnik?

File: .github/workflows/linting.yml [hunk 3]
```diff
@@ -13,6 +13,8 @@ jobs:
         uses: actions/checkout@v2
       - name: Run codegen
         run: make codegen
+      - name: Make sure there's no pending changes
```
Comment (dfarrell07) [inline .github/workflows/linting.yml:16]:
```suggestion
      - name: Verify generated code matches committed code
```

Sorry for the nit, started with avoiding "there's"->"there are".

**Review by dfarrell07 — APPROVED** [review]

**Review by mangelajo — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Pull request merged**